todoman/ui.py

```diff
@@ -258,41 +265,67 @@ def detailed(self, todo):
             rv = "{}\n\n{}".format(rv, todo.description)
         return rv
 
-    def format_date(self, date):
+    def _format_date(self, date):
         """
         Returns date in the following format:
         * if date == today or tomorrow: "Today" or "Tomorrow"
         * else: return a string representing that date
         * if no date is supplied, it returns empty_date
 
-        :param datetime.datetime date: a datetime object
+        :param datetime.date date: a date object
         """
         if date:
-            assert isinstance(date, datetime)
-            if date.date() in self.special_dates:
-                rv = self.special_dates[date.date()]
+            assert type(date) is datetime.date
+            if date in self.special_dates:
+                rv = self.special_dates[date]
             else:
                 rv = date.strftime(self.date_format)
             return rv
         else:
             return self.empty_date
 
-    def parse_date(self, date):
-        if not date:
-            return None
+    def _format_time(self, time):
+        if time:
+            assert isinstance(time, datetime.time)
+            return time.strftime(self.time_format)
+        else:
+            return self.empty_time
 
-        try:
-            rv = datetime.strptime(date, self.date_format)
-        except ValueError:
-            rv, certainty = self._parsedatetime_calendar.parse(date)
-            if not certainty:
-                raise ValueError(
-                    'Time description not recognized: {}' .format(date)
-                )
-            rv = datetime.fromtimestamp(mktime(rv))
+    def format_datetime(self, dt):
+        if not dt:
+            date_part = None
+            time_part = None
+        else:
+            assert isinstance(dt, datetime.datetime)
```

Comment:
Since we're doing this asserting, can't we do `dt.strtime(self.datetime_format)`? Can a `datetime` *not* have a time component?

To deal with `special_dates`, we can just substitute `'Tomorrow`' with '`Tomorror' + SEPARATOR + TIME_FORMAT` (at construction time).